src/main.rs

```diff
@@ -125,6 +140,49 @@ fn main() {
         startup_nsh
             .write_all(include_bytes!("startup.nsh"))
             .unwrap();
+
+        // Create user provided additional files
+        for files in additional_files {
+            let mut user_file = fs.root_dir();
+
+            let mut files = files.split(':');
+            // Safe to unwrap, there will never be an empty string passed as an arg
+            let outer = files.next().unwrap();
+
+            let inner = match files.next() {
+                // User provided an inner filename
+                Some(file) => file,
+                // No inner filename, default to the outer filename
+                // in the root dir of the image
+                None => {
+                    // Safe to unwrap because there will always be atleast
+                    // One string inside the argument
+                    outer.rsplit('/').next().unwrap()
+                }
+            };
+            let mut inner = inner.split('/');
```

Comment:
This is not a safe way to split paths. Rather use `Path::components()`.

Something like this:
```rust
let mut inner_dir_components = inner.components().take(inner.components().count() - 1);                                                                                                                
match inner_dir_components.next() {                                                                                                                                                                    
    Some(std::path::Component::RootDir) => {},                                                                                                                                                         
    _ => panic!("Invalid --add-file argument: Inner path is not absolute"),                                                                                                                            
}                                                                                                                                                                                                      
let mut current_fs_dir = fs.root_dir();                                                                                                                                                                
for inner_dir_component in inner_dir_components {                                                                                                                                                      
    match inner_dir_component {                                                                                                                                                                        
        std::path::Component::Normal(dir_osstr) => {                                                                                                                                                   
            let dir_str = dir_osstr.to_str().expect("Invalid --add-file argument: Invalid inner path encoding");                                                                                       
            current_fs_dir = current_fs_dir.create_dir(dir_str).expect("Failed to create directory on EFI image");                                                                                     
        }                                                                                                                                                                                              
        _ => panic!("Invalid --add-file argument: Invalid inner path"),                                                                                                                                
    }                                                                                                                                                                                                  
} 
```